emails/models.py

```diff
@@ -673,12 +675,16 @@ def save(self, *args, **kwargs) -> None:
                         break
                     self.address = address_default()
                 locked_profile.update_abuse_metric(address_created=True)
+                locked_profile.last_engagement = datetime.now(timezone.utc)
```

Comment:
Why update `last_engagement` here and 8 lines down?

Reply:
Oh good catch. I thought this block did a `return` so I needed to update in both places. But it looks like this block flows into the code below, so I only need to update it below.

Reply:
I kept this in because we DO want to update the field when an address is added (i.e., `self._state.adding == True`)